i3_agenda/event.py

```diff
@@ -0,0 +1,134 @@
+
+import re
+import json
+import time
+import datetime as dt
+from bidi.algorithm import get_display
+
+from typing import Optional, List
+from config import URL_REGEX
+
+
+class Event:
+    def __init__(
+        self,
+        summary: str,
+        start_time: float,
+        end_time: float,
+        location: str,
+    ):
+        self.summary = summary
+        self.start_time = start_time
+        self.end_time = end_time
+        self.location = location
+
+    def get_datetime(self):
+        return dt.datetime.fromtimestamp(self.start_time)
+
+    def get_string(self, limit_char: int, date_format: str) -> str:
+        event_datetime = self.get_datetime()
+
+        result = str(get_display(self.summary))
+
+        if 0 <= limit_char < len(result):
+            result = "".join([c for c in result][:limit_char]) + "..."
+
+        if self.is_today():
+            return f"{event_datetime:%H:%M} " + result
+        elif self.is_tomorrow():
+            return f"{event_datetime:Tomorrow at %H:%M} " + result
+        elif self.is_this_week():
+            return f"{event_datetime:%a at %H:%M} " + result
+        else:
+            return f"{event_datetime:{date_format} at %H:%M} " + result
+
+    def is_today(self):
+        today = dt.datetime.today()
+        return self.get_datetime().date() == today.date()
+
+    def is_tomorrow(self):
+        tomorrow = dt.datetime.today() + dt.timedelta(days=1)
+        return self.get_datetime().date() == tomorrow.date()
+
+    def is_this_week(self):
+        next_week = dt.datetime.today() + dt.timedelta(days=7)
+        return self.get_datetime() < next_week.date()
+
+    def is_urgent(self):
+        now = dt.datetime.now()
+        urgent = now + dt.timedelta(minutes=5)
+        five_minutes_started = self.get_datetime() + dt.timedelta(minutes=5)
+        # is urgent if it begins in five minutes and if it hasn't passed 5 minutes it started
+        return self.get_datetime() < urgent and not now > five_minutes_started
+
+    def is_allday(self) -> bool:
+        time_delta = self.end_time - self.start_time
+        return self.get_datetime().time() == dt.time(0) and time_delta % 24 == 0
```

Comment:
probably worth adding a comment on this one since it's not super clear what we're doing here